lib/gems/pending/appliance_console/internal_database_configuration.rb

```diff
@@ -126,25 +130,49 @@ def start_postgres
       block_until_postgres_accepts_connections
     end
 
+    def restart_postgres
+      LinuxAdmin::Service.new(PostgresAdmin.service_name).restart
+      block_until_postgres_accepts_connections
+    end
+
     def block_until_postgres_accepts_connections
       loop do
         break if AwesomeSpawn.run("psql -U postgres -c 'select 1'").success?
       end
     end
 
     def create_postgres_root_user
-      conn = PG.connect(:user => "postgres", :dbname => "postgres")
-      esc_pass = conn.escape_string(password)
-      conn.exec("CREATE ROLE #{username} WITH LOGIN CREATEDB SUPERUSER PASSWORD '#{esc_pass}'")
+      with_pg_connection do |conn|
+        esc_pass = conn.escape_string(password)
+        conn.exec("CREATE ROLE #{username} WITH LOGIN CREATEDB SUPERUSER PASSWORD '#{esc_pass}'")
+      end
     end
 
     def create_postgres_database
-      conn = PG.connect(:user => "postgres", :dbname => "postgres")
-      conn.exec("CREATE DATABASE #{database} OWNER #{username} ENCODING 'utf8'")
+      with_pg_connection do |conn|
+        conn.exec("CREATE DATABASE #{database} OWNER #{username} ENCODING 'utf8'")
+      end
     end
 
     def relabel_postgresql_dir
       AwesomeSpawn.run!("/sbin/restorecon -R -v #{mount_point}")
     end
+
+    def with_pg_connection
+      conn = PG.connect(:user => "postgres", :dbname => "postgres")
+      yield conn
+    ensure
+      conn.close
+    end
+
+    def apply_initial_configuration
+      shared_buffers = run_as_evm_server ? SHARED_DB_SHARED_BUFFERS : DEDICATED_DB_SHARED_BUFFERS
+      with_pg_connection do |conn|
+        conn.exec("ALTER SYSTEM SET ssl TO on") if ssl
+        conn.exec("ALTER SYSTEM SET shared_buffers TO #{shared_buffers}")
```

Comment:
As of now this is just a replacement for the old reference conf files (`EVM_V5_Dedicated_Appliance_postgresql.conf`/`EVM_V5_Shared_Appliance_postgresql.conf`).

The only difference between these and the `postgresql.conf` that was used by default was the shared_buffers setting. So now that we know if we are a dedicated database appliance or not we can just apply this directly.